documentation/developer_documentation/glossary.md

```diff
@@ -0,0 +1,78 @@
+## Algorand
+
+[Algorand](https://www.algorand.com/) is a proof-of-stake blockchain cryptocurrency protocol.
+
+## BLAKE2b
+
+BLAKE2 is a cryptographic hash function based on [BLAKE](https://en.wikipedia.org/wiki/BLAKE_(hash_function)). The design goal was to replace the widely used, but broken, MD5 and SHA-1 algorithms in applications requiring high performance in software.
+
+## BLS
+
+[BLS](https://en.wikipedia.org/wiki/BLS_digital_signature) stands for Boneh–Lynn–Shacham cryptographic signature scheme, which is a cryptographic signature scheme which allows a user to verify that a signer is authentic.
+
+## CBOR
+
+[CBOR](https://cbor.io/) stands for the Concise Binary Object Representation, which is a data format whose design goals include the possibility of extremely small code size, fairly small message size, and extensibility without the need for version negotiation.
+
+## CID
+
+[CID](https://spec.filecoin.io/#section-libraries.multiformats.cids) is short for Content Identifier, a self describing content address used throughout the IPFS ecosystem. CIDs are used in Filecoin to identify files submitted to the decentralized storage network. For more detailed information, see [the github documentation for it](https://github.com/ipld/cid).
+
+## IPLD
+
+[IPLD](https://github.com/ipld) stands for InterPlanetary Linked Data, which is a series of standards and formats for describing data in a content-addressing-emphatic way.
+
+## JWT
+
+[JWT](https://en.wikipedia.org/wiki/JSON_Web_Token) stands for JSON Web Token, which is a proposed Internet standard for creating data with optional signature and/or optional encryption whose payload holds JSON that asserts some number of claims. The tokens are signed either using a private secret or a public/private key.
+
+## SECP
+
+types of elliptic curves used for ECDSA see [here](https://www.johndcook.com/blog/2018/08/21/a-tale-of-two-elliptic-curves/)
+
+## multisig
+
+A [multi-signature](https://lotus.filecoin.io/lotus/manage/multisig/) (multisig) wallet refers to a wallet that requires multiple keys to authorize a `FIL` transactions.
+
+## Tipset
+
+Tipset is a structure that contains a non-empty collection of blocks that have distinct miners and all specify identical epoch, parents, weight, height, state root, receipt root
+
+## Tipsetkey
+
+A set of CIDs forming a unique key for a tipset.
+
+## mempool
+
+mempool stands for the Message Pool, which is the component of forest that handles pending messages for inclusion in the chain. Messages are added either directly for locally published messages or through pubsub propagation.
+
+## Merkle
+
+[Ralph C. Merkle](https://en.wikipedia.org/wiki/Ralph_Merkle) is a computer scientist. He is one of the inventors of public-key cryptography, the inventor of cryptographic hashing. Merkle tree is named after him.
```

Comment:
Best to provide a link to Merkle tree article, it's probably more interesting to someone looking into the glossary.